### PR TITLE
Fixed null pointer and deployment issue on Xenserver with L2 Guest network with configDrive

### PR DESCRIPTION
**Pull request by Spaceman1984**
## Description

This PR fixes an issue where an instance fails to deploy due to a null pointer when using an L2 Guest Network with DefaultL2NetworkOfferingConfigDrive on Xenserver. It also fixes migrating an instance to another host.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

This has been tested by: 
- Creating an L2 Guest network, using DefaultL2NetworkOfferingConfigDrive as the network offering. 
- Deploying an instance using the L2 Guest network created.
- Migrating the instance away from the host and back

### CODE REVIEW

**Comment by Spaceman1984** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test centos7 xenserver-65sp1

File: plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java [hunk 1]
```diff
@@ -1156,7 +1156,7 @@ public VBD createVbd(final Connection conn, final DiskTO volume, final String vm
         if (volume.getType() == Volume.Type.ISO) {
             vbdr.mode = Types.VbdMode.RO;
             vbdr.type = Types.VbdType.CD;
-            vbdr.userdevice = "3";
+            vbdr.userdevice = String.valueOf(3 + isoCount);
```
Comment (DaanHoogland) [inline plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java:1159]: maybe give the 3 a constant with descriptive name

Comment (Spaceman1984) [reply plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java]: @DaanHoogland I have made the changes.

File: plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java [hunk 2]
```diff
@@ -5542,7 +5544,7 @@ public boolean attachConfigDriveToMigratedVm(Connection conn, String vmName, Str
             s_logger.debug("Attaching config drive iso device for the VM " + vmName + " In host " + ipAddr);
             Set<VM> vms = VM.getByNameLabel(conn, vmName);
 
-            SR sr = getSRByNameLabel(conn, _configDriveSRName + ipAddr);
+            SR sr = getSRByNameLabel(conn, vmName + "-ISO");
```
Comment (DaanHoogland) [inline plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java:5547]: ¿can you create a final (static) field for this?

Comment (Spaceman1984) [reply plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java]: Done

**Comment by Spaceman1984** [comment]
@blueorangutan package

**Comment by Spaceman1984** [comment]
@blueorangutan test centos7 xenserver-65sp1

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 xenserver70

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 xenserver-71

File: server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java [hunk 3]
```diff
@@ -478,7 +476,7 @@ private boolean deleteConfigDriveIso(final VirtualMachine vm) throws ResourceUna
         return true;
     }
 
-    private void addConfigDriveDisk(final VirtualMachineProfile profile, final DataStore dataStore) throws ResourceUnavailableException {
+    public static void addConfigDriveDisk(final VirtualMachineProfile profile, final DataStore dataStore) throws ResourceUnavailableException {
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java:479]: can you explain? I don't see it being used anywhere

Comment (nvazquez) [reply server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java:479]: @Spaceman1984 have you seen this question?

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test centos7 xenserver-71

File: server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java (hunk 3 above)
Comment (Spaceman1984) [reply server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java:479]: Must be a leftover artifact from long days of experimentation, will remove.

**Comment by rhtyd** [comment]
@DaanHoogland can you review this? Thanks (I'm not following this)

**Comment by DaanHoogland** [comment]
@nvazquez (cc @Spaceman1984 ) I'm leaving this with you for now? it was reviewed and tested. Can you give it a last glance and merge?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by rhtyd** [comment]
@Spaceman1984 cc @nvazquez please advise, if you've tested then we can merge this

**Comment by Spaceman1984** [comment]
I didn't test multiple network interfaces.

**Comment by rhtyd** [comment]
@borisstoyanov per your testing is this ready to be merged, or needs further testing, the case of multiple nics per Daan's comment?

**Comment by rhtyd** [comment]
Discussed with @Pearl1594, the operations for config drive are only consumed on the default nics; in case of multiple nics there won't be any regressions.

**Pull request merged**